### PR TITLE
Feature 1578 upgrade scancode

### PR DESCRIPTION
**Pull request by Jeeppler**
- Upgrade scancode
- Use PDS base image
- Use messages for diagnostics
- Overwrite `pool.py` (ugly hack; see: https://github.com/nexB/scancode-toolkit/issues/2908)
- Smaller fixes

closes: #1578
closes: #1633

### CODE REVIEW

**Review by sven-dmlr — CHANGES_REQUESTED** [review]
Looks overall good. :+1: 

Please check the comments.

File: sechub-pds-solutions/scancode/docker/scripts/scancode.sh [hunk 1]
```diff
@@ -130,21 +177,23 @@ if [[ "$extractcode_enabled" == "true" ]]
...
 
 echo ""
 echo "----------------"
 echo "Preparing result"
 echo "----------------"
 echo ""
 
+if [[ "$SCANCODE_SCAN_DIAGNOSTICS" == "true" ]]
+then
+    mv "$diagnostics_file" "${PDS_JOB_USER_MESSAGES_FOLDER}/INFO_message_$(date +%Y-%m-%d_%H.%M.%S_%N).txt"
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/docker/scripts/scancode.sh:194]: You are spot, on the file is huge.

File: sechub-pds-solutions/scancode/docker/scripts/scancode_mock.sh [hunk 2]
```diff
@@ -21,7 +21,25 @@ case "$given_output_format" in
...
+echo "- scancode.scan.diagnostics: $SCANCODE_SCAN_DIAGNOSTICS"
 echo "Given output format: $given_output_format"
 echo "Mock file: $mock_file"
-cp "$mock_file" "$PDS_JOB_RESULT_FILE"
\ No newline at end of file
+cp "$mock_file" "$PDS_JOB_RESULT_FILE"
+
+if [[ "$SCANCODE_SCAN_DIAGNOSTICS" == "true" ]]
+then
+    cat "$MOCK_FOLDER/scancode_mock.json" > "${PDS_JOB_USER_MESSAGES_FOLDER}/INFO_message_$(date +%Y-%m-%d_%H.%M.%S_%N).txt"
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/docker/scripts/scancode_mock.sh:44]: Same reason, the file is very large.

File: sechub-pds-solutions/scancode/test/pds-job-config.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+    "apiVersion" : "1.0",
+    "sechubJobUUID": "288607bf-ac81-4088-842c-005d5702a9e9", 
+    "productId": "PDS_SCANCODE",
+    "parameters": [
+        {
+            "key": "pds.scan.configuration",
+            "value": "{\n  \"data\" : {\n    \"sources\" : [ ],\n    \"binaries\" : [ {\n      \"fileSystem\" : {\n        \"files\" : [ \"Armbian_22.08.1_Jetson-nano_jammy_current_5.18.19.img\" ],\n        \"folders\" : [ ]\n      },\n      \"name\" : \"test-file\"\n    } ]\n  },\n  \"licenseScan\" : {\n    \"use\" : [ \"test-file\" ]\n  },\n  \"apiVersion\" : \"1.0\"\n}"
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/test/pds-job-config.json:8]: I will keep this for now. This is output generated by the PDS-Tool.

File: sechub-pds-solutions/scancode/docker/scripts/message.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+
+# TODO: this script can be moved to the PDS base image
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-solutions/scancode/docker/scripts/message.sh:4]: Done.

**Review by sven-dmlr — APPROVED** [review]

**Pull request merged**